docs/serial_driver.md

```diff
@@ -68,3 +71,140 @@ You must also enable the ChibiOS `SERIAL` feature:
 * In your board's mcuconf.h: `#define STM32_SERIAL_USE_USARTn TRUE` (where 'n' matches the peripheral number of your selected USART on the MCU)
 
 Do note that the configuration required is for the `SERIAL` peripheral, not the `UART` peripheral.
+
+### USART Full-duplex
+Targeting STM32 boards where communication is offloaded to a USART hardware device. The advantage over bitbang is that this provides fast and accurate timings. USART Full-Duplex requires two conductors instead of one conductor unlike the Half-duplex driver, but it is more efficent as it uses DMA transfers, which can result in even faster transmission speeds.
+
+#### Pin configuration
+
+`SERIAL_USART_TX_PIN` is the USART `TX` pin, `SERIAL_USART_RX_PIN` is the USART `RX` pin. No external pull-up resistors are needed as the `TX` pin operates in push-pull mode. To use this driver the usart peripherals `TX` and `RX` pins must be configured with the correct Alternate-functions. If you are using an STM32F103 everything is already setup, for other MCUs which are using a more modern GPIO configuration you have to specify these by setting `SERIAL_USART_TX_PAL_MODE` and `SERIAL_USART_RX_PAL_MODE`. Refeer to the corresponding datasheets of your MCU or find those Settings in the table below.
```

Comment:
Pretty sure we target Proton-C/F303 by default. Will need to confirm defaults elsewhere.
```suggestion
`SERIAL_USART_TX_PIN` is the USART `TX` pin, `SERIAL_USART_RX_PIN` is the USART `RX` pin. No external pull-up resistors are needed as the `TX` pin operates in push-pull mode. To use this driver the usart peripherals `TX` and `RX` pins must be configured with the correct Alternate-functions. If you are using a Proton-C, everything is already setup, for other MCUs which are using a more modern GPIO configuration you have to specify these by setting `SERIAL_USART_TX_PAL_MODE` and `SERIAL_USART_RX_PAL_MODE`. Refeer to the corresponding datasheets of your MCU or find those Settings in the table below.
```

Reply:
Added Proton-C to the sentence and adjusted the following, not mentioning the stm32f103 would have skewed the meaning as If the f303 wouldn't have a modern gpio configuration.